sdks/python/apache_beam/typehints/decorators.py

```diff
@@ -269,6 +272,14 @@ def getcallargs_forhints(func, *typeargs, **typekwargs):
                      for (arg, hint) in zip(argspec.args, typeargs)]
   packed_typeargs += list(typeargs[len(packed_typeargs):])
 
+  if sys.version_info.major < 3:
+    return getcallargs_forhints_impl_py2(func, argspec, packed_typeargs,
+                                         typekwargs)
+  else:
+    return getcallargs_forhints_impl_py3(func, packed_typeargs, typekwargs)
+
+
+def getcallargs_forhints_impl_py2(func, argspec, packed_typeargs, typekwargs):
```

Comment:
If we separate py2 and py3 codepath here, we should be able to remove `if _use_full_argspec:` branch from getcallargs_forhints_impl_py2.